tests/lib/rules/no-timers.js

```diff
@@ -0,0 +1,96 @@
+const rule = require('../../../lib/rules/no-timers');
+const MESSAGE = rule.meta.message;
+const RuleTester = require('eslint').RuleTester;
+const ruleTester = new RuleTester();
+const parserOptions = {
+  ecmaVersion: 6,
+  sourceType: 'module'
+};
+
+ruleTester.run('no-timers', rule, {
+  valid: [
+    {
+      code: `
+        export default Ember.Component.extend({
+          actions: {
+            foo() {
+              setTimeout(() => {});
+            }
+          }
+        });`,
+      parserOptions
+    },
+    {
+      code: `
+        export default Ember.Component.extend({
+          guarded() {
+            if (environment.isBrowser()) {
+              setTimeout(() => {});
+            }
+          }
+        });`,
+      parserOptions
+    },
+    {
+      code: `
+        export default Ember.Component.extend({
+          guarded() {
+            if (environment.isBrowser()) {
+              setInterval(() => {});
```

Comment:
So this rule can still be relevant, but just not included as part of the recommended rules. We'd like this to be more of a toolkit that people can pick and choose rules from.

To that end, I'm still going to include this as a valid test. We can adjust later as things change.